src/internal/operations/common.go

```diff
@@ -20,18 +21,23 @@ func getBackupAndDetailsFromID(
 	detailsStore streamstore.Reader,
 	errs *fault.Bus,
 ) (*backup.Backup, *details.Details, error) {
-	dID, bup, err := ms.GetDetailsIDFromBackupID(ctx, backupID)
+	bup, err := ms.GetBackup(ctx, backupID)
 	if err != nil {
 		return nil, nil, errors.Wrap(err, "getting backup details ID")
 	}
 
 	var (
-		deets details.Details
-		umt   = details.UnmarshalTo(&deets)
+		deets     details.Details
+		umt       = details.UnmarshalTo(&deets)
+		detailsID = bup.DetailsID
 	)
 
-	if err := detailsStore.Read(ctx, dID, umt, errs); err != nil {
-		return nil, nil, errors.Wrap(err, "getting backup details data")
+	if len(detailsID) == 0 {
+		return bup, nil, clues.New("no details in backup").WithClues(ctx)
```

Comment:
Since the backupID (and a ctx) gets passed in by a caller, this code assumes that it's already stashed in the CTX.  I'll double check that it is, just in case.

Reply:
Yup: it's there